packages/beacon-node/src/api/impl/beacon/blocks/index.ts

```diff
@@ -407,16 +407,22 @@ export function getBeaconBlockApi({
 
     async getBlockAttestations({blockId}) {
       const {block, executionOptimistic, finalized} = await getBlockResponse(chain, blockId);
+      const fork = config.getForkName(block.message.slot);
+
+      if (isForkPostElectra(fork)) {
```

Comment:
We need to apply similar logic to other apis that only support phase0 once https://github.com/ChainSafe/lodestar/pull/6998 is merged

Reply:
After Electra is live, we should probably hardcode every deprecated attestation V1 endpoints to throw deprecation error ie. get rid of this `if (isForkPostElectra(fork))` statement

Reply:
Yes, we can just statically throw an error and refer to v2 apis similar to this error. After electra is live we can also hard-code everywhere to use v2 apis only on the validator client side as all beacon node must have implemented them at this point, otherwise they can't handle electra attestations.